_data/contributors.js

```diff
@@ -3,81 +3,91 @@ const { AssetCache } = require('@11ty/eleventy-fetch');
 const { graphql } = require('@octokit/graphql');
 
 const ORGANIZATION_NAME = 'RedHat-Israel'; // organization namespace
-const MAX_CONTRIBUTORS = 100; // modify this if/when we have more then 100 members
+const MAX_CONTRIBUTORS_FETCH = 20;
+// REQUIRED: set token from a Secret into the SITE_GITHUB_TOKEN environment variable in the CI workflow
+// token requirements: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#authenticating-with-graphql
+const REQUEST_PARAMS = {
+  headers: {
+    authorization: `bearer ${process.env.SITE_GITHUB_TOKEN}`,
+  }
+};
+
+
+module.exports = israelContributors;
 
 /**
- * Fetch Contributor data from GitHub and elsewhere
+ * Fetch Contributors data from GitHub
  * @param {*} configData see https://www.11ty.dev/docs/data-js/#arguments-to-global-data-files
- * @example
- *     ```html
- *     <ul>{% for contributor in contributors %}
- *       <li>{{ contributor.name }}</li>{% endfor %}
- *     </ul>
- *     ```
  */
-async function israelContributors(configData) {
-  // REQUIRED: set token from a Secret into the SITE_GITHUB_TOKEN environment variable in the CI workflow
-  // token requirements: https://docs.github.com/en/graphql/guides/forming-calls-with-graphql#authenticating-with-graphql
-  const requestParams = {
-    headers: {
-      authorization: `bearer ${process.env.SITE_GITHUB_TOKEN}`,
-    }
+async function israelContributors(_configData) {
+  const cache = new AssetCache('github_graphql_contributions');
+  const summary = cache.isCacheValid('1d') ? cache.getCachedValue() : await getStatsForOrgMembers(`first: ${MAX_CONTRIBUTORS_FETCH}`);
+  return {
+    ...summary,
   };
+}
 
+async function getStatsForOrgMembers(members, summary) {
+  if (!summary) {
+    summary = {
+      totalCommitContributions: 0,
+      totalIssueContributions: 0,
+      totalPullRequestContributions: 0,
+      totalPullRequestReviewContributions: 0,
+      totalRepositoryContributions: 0,
+      totalGists: 0
+    };
+  }
+
+  // NOTE: fetching "max-per-page" 100 gists per member
   const query = `
     {
       organization (login: "${ORGANIZATION_NAME}") {
-        membersWithRole(first: ${MAX_CONTRIBUTORS}) {
-          nodes {
-            contributionsCollection {
-              hasAnyContributions
-              totalCommitContributions
-              totalIssueContributions
-              totalPullRequestContributions
-              totalPullRequestReviewContributions
-              totalRepositoryContributions
-            }
-            gists (first: 100) {
-              totalCount
+        membersWithRole(${members}) {
```

Comment:
now that we're here, perhaps this can be refactored to use GQL [variables](https://github.com/octokit/graphql.js#variables)? Then we could write this query in a .gql file instead of a template literal

Reply:
@bennypowers 
Ok, I get the idea of using variables instead of dynamic queries to avoid injection attacks.
I'll take care of that, but I'm not following you on the _gql_ file, can you elaborate on that?

Reply:
```ts
const query = await readFile(new URL('./contributors.query.gql', import.meta.url), 'utf8');

const result = await graphql({ query, variables, headers });
```

Reply:
writing gql in gql has a few advantages:
- syntax highlighting
- language server
- linting
- maintenance

Reply:
actually...
using variables will not work for us.

in the link you provided, the maintainer references the issue that caused him to recommend using variables instead of template literals, note his comment here, https://github.com/octokit/graphql.js/issues/2#issuecomment-445303444, and the last comment on the issue, he didn't do anything further then recomendending using variables, I guess dynamic queries has a more common usage.

we use _dynamic queries_, as he suggested, note this:
https://github.com/RedHat-Israel/red-hat-israel-site/blob/f9f0613f641733022c5f626deda1119bb3651693/_data/contributors.js#L24

and this:
https://github.com/RedHat-Israel/red-hat-israel-site/blob/f9f0613f641733022c5f626deda1119bb3651693/_data/contributors.js#L89

for each case, we pass a different set of filter properties, we do not have the `after` cursor value for the first run, just of the next runs, so using variables will not quite fit here I believe.

WDYT?

Reply:
> ```ts
> const query = await readFile(new URL('./contributors.query.gql', import.meta.url), 'utf8');
> 
> const result = await graphql({ query, variables, headers });
> ```

thanks... I was wondering if js had the ability to read files. 😜 

> syntax highlighting
> language server
> linting
> maintenance

nice, it's a relatively small query so it's not such a big deal, but I'll look into it.
thanks.

Reply:
@bennypowers 
I pushed a new commit to demonstrate my point, I can revert it if you don't like it.

1. I got to admit, I like the idea of using variables instead of template literals, it makes the code a bit more readable and cleaner I think.

2. in regards to, _syntax highlighting_, _language server_, and _linting_, I'm using _vscode_ with [this graphql extension](https://marketplace.visualstudio.com/items?itemName=GraphQL.vscode-graphql), turns out I simply needed to add a comment "_#graphql_" in the query, and I got everything working. I can still move the queries into _gql_ files if you prefer it that way.

3. in regards to _maintenance_, as explained above, we now have two "almost-identical" queries to maintain, if we opt to outsource them into _gql_ files, then we have to keep it that way, if we opt to keep them as strings, then I can perhaps play around with them a bit and combined them into 1 query concatenated at runtime, but then we're back where we started, template literals.

Please advise on the next step,
I'm ok with leaving this as it is now (using variables + 2 queries).

Reply:
BTW

note that I tweaked the max members fetch from _20_ to _15_,
I was starting to view timeouts with 20 as well.

Reply:
In the last commit, I moved the shared attributed for both queries in a fragment, which reduces the maintenance levels back down.

Reply:
Yeah I was thinking a fragment would be good
Might require some trickery to get it into a gql file, which I still think is better then template literals, but if you're itching to merge we can do that later

Reply:
I did the fragment thing in the last commit, only I did it using variables and not files.
I think it looks good, we have 2 small "query" variables leveraging 1 "fragment" variable.

I think it's good, check it out: https://github.com/RedHat-Israel/red-hat-israel-site/blob/fix-contributors-query-timeouts/_data/contributors.js#L16

Reply:
jumped in with a few small changes, namely file loading. its [not ideal](https://github.com/octokit/graphql.js/issues/331) but its ok for now